roxie/ccd/ccdfile.cpp

```diff
@@ -1270,6 +1616,211 @@ class CRoxieFileCache : implements IRoxieFileCache, implements ICopyFileProgress
+                if (file_mmap == MAP_FAILED)
+                {
+                    DBGLOG("Failed to map file %s to pre-warm cache (error %d)", fileName.str(), errno);
+                    file_mmap = nullptr;
+                }
+            }
+            else if (traceLevel)
+            {
+                DBGLOG("Failed to open file %s to pre-warm cache (error %d)", fileName.str(), errno);
+            }
+            Owned<ILazyFileIO> localFile = lookupLocalFile(fileName);
+            if (localFile)
+            {
+                fileIdx = localFile->getFileIdx();
+            }
+            cacheInfo = endName+1;  // Skip the |
+            while (*cacheInfo==' ')
+                cacheInfo++;
+            for (;;)
+            {
+                bool inNodeCache = (*cacheInfo=='*');
+                if (inNodeCache)
+                    cacheInfo++;
+                offset_t startPage = readPage(cacheInfo);
+                offset_t endPage;
+                if (*cacheInfo=='-')
+                {
+                    cacheInfo++;
+                    endPage = readPage(cacheInfo);
+                }
+                else
+                    endPage = startPage;
+                if (traceLevel > 4)
+                    DBGLOG("Touching %s %" I64F "x-%" I64F "x", fileName.str(), startPage, endPage);
+                startPage = startPage << CacheInfoEntry::pageBits;
+                endPage = (endPage+1) << CacheInfoEntry::pageBits;
+                if (inNodeCache && !keyFailed && localFile && !keyIndex)
+                {
+                    keyIndex.setown(createKeyIndex(fileName, localFile->getCrc(), *localFile.get(), fileIdx, false, false));  // MORE - we don't know if it's a TLK, but hopefully it doesn't matter
+                    if (!keyIndex)
+                        keyFailed = true;
+                }
+                if (inNodeCache && keyIndex)
+                {
+                    // Round startPage up to nearest multiple of node size
+                    unsigned nodeSize = keyIndex->getNodeSize();
+                    startPage = ((startPage+nodeSize-1)/nodeSize)*nodeSize;
+                    do
+                    {
+                        bool loaded = keyIndex->prewarmPage(startPage);
+                        if (!loaded)
+                            break;
+                        preloaded++;
+                        startPage += nodeSize;
+                    }
+                    while (startPage < endPage);
+                }
+                else if (file_mmap)
+                {
+                    if (fileIdx != (unsigned) -1)
+                        noteRead(fileIdx, startPage, (endPage-1) - startPage);
+                    do
+                    {
+                        if (startPage >= (offset_t) file_stat.st_size)
+                            break;    // Let's not core if the file has changed size since we recorded the info...
+                        t += file_mmap[startPage];
```

Comment:
Add a comment about it being reported to ensure it is not optimized away.  Originally I thought it might be.

Reply:
I did, on the line that reports it!